executor/aggfuncs/func_count_distinct.go

```diff
@@ -242,6 +243,8 @@ func (e *countOriginalWithDistinct4String) AppendFinalResult2Chunk(sctx sessionc
 
 func (e *countOriginalWithDistinct4String) UpdatePartialResult(sctx sessionctx.Context, rowsInGroup []chunk.Row, pr PartialResult) error {
 	p := (*partialResult4CountDistinctString)(pr)
+	_, coll := e.args[0].CharsetAndCollation(sctx)
```

Comment:
We should use the result type of the first argument directly.

Reply:
You mean that we use `e.args[0].RetType.Collation` here?

Reply:
Yeah

Reply:
updated, PTAL.